src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -551,3 +551,24 @@ function rrule(::typeof(getproperty), F::T, x::Symbol) where {T <: Cholesky}
     end
     return getproperty(F, x), getproperty_cholesky_pullback
 end
+
+# `det` and `logdet` for `Cholesky`
+function rrule(::typeof(det), C::Cholesky)
+    y = det(C)
+    s = conj!((2 * y) ./ _diag_view(C.factors))
+    function det_Cholesky_pullback(ȳ)
+        ΔC = Tangent{typeof(C)}(; factors=Diagonal(ȳ .* s))
```

Comment:
Would something like this be better? There's one fewer allocation, and we don't need to assume that `s` is mutable.
```suggestion
   diagF = _diag_view(C.factors)
    function det_Cholesky_pullback(ȳ)
        ΔC = Tangent{typeof(C)}(; factors=Diagonal(2(ȳ * conj(y)) ./ conj.(diagF)))
```

Reply:
If the determinant is 0 (can happen if `check=false` is passed to `cholesky`), this will inject `NaN`s, even if the cotangent is 0. Since we try to treat cotangents as strong zeros, it would be nice to handle this case by ensuring that such `NaN`s end up as zeros.

Reply:
> There's one fewer allocation, and we don't need to assume that s is mutable.

Seems like an improvement to me, thanks! :slightly_smiling_face: 

There's a whitespace missing in the first line of your suggestion it seem:

```suggestion
    diagF = _diag_view(C.factors)
    function det_Cholesky_pullback(ȳ)
        ΔC = Tangent{typeof(C)}(; factors=Diagonal(2(ȳ * conj(y)) ./ conj.(diagF)))
```

Reply:
> If the determinant is 0

It would happen if `y = 0` (the determinant) but also if `ȳ = 0`. Should we care about the last case as well? Or is it correct to return `NaN` there?

Reply:
Ah, never mind, of course, at least one element of `diagF` is zero iff`y = 0`. I.e., we only have to care about `y = 0`.

Reply:
Maybe even a bit clearer (without having to know about precedence of operators):
```suggestion
    diagF = _diag_view(C.factors)
    function det_Cholesky_pullback(ȳ)
        ΔC = Tangent{typeof(C)}(; factors=Diagonal( (2 * (ȳ * conj(y))) ./ conj.(diagF)))
```

Reply:
I guess something like the following could work?
```julia
# compute `x / conj(y)`, handling `x = y = 0`
function _x_divide_conj_y(x, y)
    z = x / conj(y)
    # in our case `iszero(x)` implies `iszero(y)`
    return iszero(x) ? zero(z) : z
end
function rrule(::typeof(det), C::Cholesky)
    y = det(C)
    diagF = _diag_view(C.factors)
    function det_Cholesky_pullback(ȳ)
        ΔC = Tangent{typeof(C)}(; factors=Diagonal( _x_divide_conj_y(2 * (ȳ * conj(y)), diagF) ))
```

Reply:
@sethaxen I updated the PR and added tests for singular matrices.